packages/menu-bar/test/keyboard-navigation.test.js

```diff
@@ -138,6 +138,62 @@ describe('keyboard navigation', () => {
           expect(buttons[3].hasAttribute('focused')).to.be.true;
         });
       });
+
+      describe('overflow', () => {
+        beforeEach(async () => {
+          menu.items = [{ text: 'Item Foo' }, { text: 'Item Bar' }];
+          await nextRender();
+          buttons = menu._buttons;
+        });
+
+        it('should move focus back to the overflow button on Shift + Tab after Tab', async () => {
+          // Hide all buttons except overflow
+          menu.style.maxWidth = '100px';
+          await nextResize(menu);
+
+          firstGlobalFocusable.focus();
+          await sendKeys({ press: 'Tab' });
+          expect(document.activeElement).to.equal(menu._overflow);
+
+          await sendKeys({ press: 'Tab' });
+          expect(document.activeElement).to.equal(lastGlobalFocusable);
+
+          await sendKeys({ press: 'Shift+Tab' });
+          expect(document.activeElement).to.equal(menu._overflow);
+        });
+
+        it('should move focus back to the overflow button on Tab after Shift + Tab', async () => {
+          // Show 1 button + overflow
+          menu.style.maxWidth = '120px';
+          await nextResize(menu);
+
+          lastGlobalFocusable.focus();
+          await sendKeys({ press: 'Shift+Tab' });
+          expect(document.activeElement).to.equal(menu._overflow);
```

Comment:
In this case, the focused button depends on where you focus from, because both visible buttons have tabindex 0.
Effectively, whenever you resize the menubar, the overflow button will always get tabindex 0 as long as it's visible, and will then get shift-focused to even though the focus was previously on some other button.

Reply:
```js
it('should preserve focusable button after resize', async () => {
  // Show 1 button + overflow
  menu.style.maxWidth = '120px';
  await nextResize(menu);

  firstGlobalFocusable.focus();

  // Tab to the first button
  await sendKeys({ press: 'Tab' });
  expect(document.activeElement).to.equal(buttons[0]);

  lastGlobalFocusable.focus();

  // Resize slightly, should have no effect on overflowed buttons
  menu.style.maxWidth = '121px';
  await nextResize(menu);

  await sendKeys({ press: 'Shift+Tab' });
  expect(document.activeElement).to.equal(buttons[0]);
});
```

Reply:
Restored the original fix and adapted `tabindex` logic to only set to `0` when overflow is the only visible button.